lib/routing.py

```diff
@@ -127,33 +137,48 @@ def run(self, argv=None):
         self.path = self.path.rstrip('/')
         if len(argv) > 2:
             self.args = parse_qs(argv[2].lstrip('?'))
+            self.args = dict((k, list(uq(uq(v2)) for v2 in v)) for k, v in self.args.items())
```

Comment:
Why the double quoting and unquoting?
We need to document the reason for added complexity.

Reply:
I put it in the test for some reason....idk it was late.
```
# we wanted double quote for the +, %, and any others that might be in the string
```
It won't touch ASCII, but double unquoting doesn't hurt if there's a chance of invalid parsing.

Reply:
But why would there be any invalid parsing? I would like to see concrete examples where the current implementation failed.